nixos/modules/flyingcircus/roles/webproxy.nix

```diff
@@ -40,12 +52,30 @@ in
          (fclib.listenAddressesQuotedV6 config "lo"));
     services.varnish.config = varnishCfg;
     services.varnish.stateDir = "/var/spool/varnish/${config.networking.hostName}";
-    # XXX: needs to be migrated to varnish service
-    systemd.services.varnish.after = [ "network.target" ];
+    systemd.services.varnish = {
+      # XXX: needs to be migrated to varnish service
+      after = [ "network.target" ];
+      preStart = lib.mkAfter ''
+        install -d -o ${toString config.ids.uids.varnish} -g service -m 02775 /etc/local/varnish
+      '';
+      reloadIfChanged = true;
+      restartTriggers = [ configFile ];
+      reload = ''
+        if ${pkgs.procps}/bin/pgrep -a varnish | grep  -Fq '${varnish_}'
+        then
+          config=$(readlink -e /etc/varnish.vcl)
+          ${pkgs.varnish}/bin/varnishadm vcl.load $config $config &&
+            ${pkgs.varnish}/bin/varnishadm vcl.use $config
+        else
+          echo "Binary or parameters changed. Restarting."
+          systemctl restart varnish
+        fi
+      '';
```

Comment:
The script's readability could benefit from populating the `path` variable with all required packages.

Reply:
I agree but would like to say that isn't release blocking in any way.